### DISCUSSION TITLE
Specifiying the input language

### DISCUSSION THREAD

**Discussion by yilmazay74** [mallorbc/whisper_mic#60] [Ideas]
Hi All,
When I am trying whisper_mic, I see that it is biased towards English. When I run it without --english flag it supposed to recognize speaker's language and decode in that language, right?
However, during my tests, usually it tries to decode in English, although I want to test it in Turkish. 
It can hardly decode a few words in Turkish, most of them in English and most other times in other languages.
So, I mean, the auto language recognition is not working properly.
Actually in normal whisper we can specify the language so that it decodes in that language. And it really makes a lot of difference in both speed and accuracy. Although it would be ok even if could correctly detect the language automatically, but usually it can't. 
So, I would suggest to add an input --lang parameter to force it decode in that language only.
What do you think guys?

**Comment by yilmazay74** [comment]
Hi Everyone,
When I looked at the whisper_mic.py code, I found the place where the language is set.
It is between 176 and 179.
Just change it as follows:
The original code snippet:       
 _if self.english:
            result = self.audio_model.transcribe(audio_data,language='english')
        else:
            result = self.audio_model.transcribe(audio_data)_
The modified code snippet that decodes in the specified language:
        _if self.english:
            result = self.audio_model.transcribe(audio_data,language='english')
        else:
            result = self.audio_model.transcribe(audio_data, **language='turkish'**)_

Hope, it helps someone who is facing a similar problem.
Regards

**Reply by Ayushhkkux** [reply]
Thank you so much .... I was in need of this ... for hindi language... for my speech emotion recognition....

**Comment by mallorbc** [comment]
Ah yes.  Currently, the model will be expecting english input or will be looking to convert the spoken language to english.  This is a good find.  If you make a PR I will look to merge it.

Thanks!

**Reply by Ayushhkkux** [reply]
Is it possible to live transcribe the classifier model also ... to detect the emotion from live speeches?? Pls tell

**Comment by Ayushhkkux** [comment]
> Hi All, When I am trying whisper_mic, I see that it is biased towards English. When I run it without --english flag it supposed to recognize speaker's language and decode in that language, right? However, during my tests, usually it tries to decode in English, although I want to test it in Turkish. It can hardly decode a few words in Turkish, most of them in English and most other times in other languages. So, I mean, the auto language recognition is not working properly. Actually in normal whisper we can specify the language so that it decodes in that language. And it really makes a lot of difference in both speed and accuracy. Although it would be ok even if could correctly detect the language automatically, but usually it can't. So, I would suggest to add an input --lang parameter to force it decode in that language only. What do you think guys?

By the way I did the adjustment on (cli.py )...... 
As it looks so messy to me with a large lines of code.... I saw 35 to 36 line of code there to transcribe. Is it okay??
